pkg/trait/knative.go

```diff
@@ -81,6 +83,24 @@ func (t *knativeTrait) Configure(e *Environment) (bool, *TraitCondition, error)
 		if err != nil {
 			return false, nil, err
 		}
+
+		if t.Enabled == nil {
```

Comment:
I think this is what the different `filterMetaItems` calls after this piece of code do. At the end of the `configure` you should have all the knative endpoints in the ChannelSources, ChannelSinks, ... variable. I think it is not convenient calculating this twice. Maybe, you can check those resources after they have been collected and do the needed logic accordingly.

Reply:
I noticed it, but wanted to have a check before, as these filters are intended to check for the specifics of knative. Anyway, I changed the check to account for the results of the knative endpoints.